### PR TITLE
Add categories and shortcuts to _WKApplicationManifest

### PR DESCRIPTION
**Pull request by LumingYin**
#### a4ea62848c891ee3c0aaff18417e2cacfc3d064c
<pre>
Add categories and shortcuts to _WKApplicationManifest
<a href="https://bugs.webkit.org/show_bug.cgi?id=262654">https://bugs.webkit.org/show_bug.cgi?id=262654</a>
rdar://116486952

Reviewed by Wenson Hsieh, Sihui Liu and Chris Dumez.

Add two new properties, categories and shortcuts, to _WKApplicationManifest.
See rdar://116486952 for more detail.

* Source/WebCore/Modules/applicationmanifest/ApplicationManifest.h:
Add a new Shortcut struct that represents an individual shortcut, as well
as categories and shortcuts members on the manifest.

* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp:
(WebCore::ApplicationManifestParser::parseManifest):
Parse the categories and shortcuts members of the manifest.

(WebCore::ApplicationManifestParser::parseCategories):
Parse categories by mapping its values into a vector of strings.

(WebCore::ApplicationManifestParser::parseShortcuts):
Parse shortcuts by mapping its values into a vector of shortcut items. For each
icon member inside each shortcut, use the existing parseIcons method to parse
them since they are in the same format as the top-level icons member.

* Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.h:
Header declaration for new methods that parse categories and shortcuts.

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
Serialize the parsed categories and shortcuts for XPC.

* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:
(makeVectorElement):
Method to convert an individual array element of _WKApplicationManifestShortcut
back into WebCore::ApplicationManifest::Shortcut. This is used when initializing
with coder. Remove the isKindOfClass check since dynamic_objc_cast already returns
nil if the class check fails.

(-[_WKApplicationManifestIcon initWithCoder:]):
(-[_WKApplicationManifestIcon initWithCoreIcon:]):
(-[_WKApplicationManifestIcon encodeWithCoder:]):
(-[_WKApplicationManifestIcon src]):
(-[_WKApplicationManifestIcon sizes]):
(-[_WKApplicationManifestIcon type]):
(-[_WKApplicationManifestIcon purposes]):
(-[_WKApplicationManifestIcon dealloc]):
Switch to using RetainPtr.

(+[_WKApplicationManifestShortcut supportsSecureCoding]):
Allow secure encoding of _WKApplicationManifestShortcut.

(-[_WKApplicationManifestShortcut initWithCoder:]):
Decode an encoded _WKApplicationManifestShortcut.

(-[_WKApplicationManifestShortcut name]):
(-[_WKApplicationManifestShortcut url]):
(-[_WKApplicationManifestShortcut icons]):
Getter for RetainPtr backed ivars.

(-[_WKApplicationManifestShortcut initWithCoreShortcut:]):
Initialize Obj-C wrapper object based on the WebCore shortcut object.

(-[_WKApplicationManifestShortcut encodeWithCoder:]):
Encode _WKApplicationManifestShortcut for on-disk persistence.

(-[_WKApplicationManifestShortcut dealloc]):
Deallocate _WKApplicationManifestShortcut.

(-[_WKApplicationManifest initWithCoder:]):
Decode categories and shortcuts.

(-[_WKApplicationManifest encodeWithCoder:]):
Encode categories and shortcuts.

(-[_WKApplicationManifest categories]):
Return categories as an array of NSString from the underlying manifest.

(-[_WKApplicationManifest shortcuts]):
Return shortcuts as an array of _WKApplicationManifestShortcut from the
underlying manifest.

* Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp:
(ApplicationManifestParserTest::parseShortcutFirstTopLevelProperty):
(ApplicationManifestParserTest::parseShortcutFirstTopLevelPropertyForURL):
Test whether shortcut property can be parsed correctly.

(ApplicationManifestParserTest::parseShortcutIconFirstTopLevelProperty):
(ApplicationManifestParserTest::parseShortcutIconFirstTopLevelPropertyForSrc):
Test whether an icon nested within a shortcut can be parsed correctly.

(ApplicationManifestParserTest::testCategories):
Test whether a set of categories can be parsed correctly.

(ApplicationManifestParserTest::testIconsSrc):
(ApplicationManifestParserTest::testIconsType):
(ApplicationManifestParserTest::testIconsSizes):
(ApplicationManifestParserTest::testIconsPurposes):
Update icon test methods to also test icons nested within shortcuts members.

(ApplicationManifestParserTest::testShortcutsURL):
(ApplicationManifestParserTest::testShortcutsName):
Test shortcuts url and name.

(TEST_F):
Run categories and shortcuts tests. Also make a drive-by test fix for Scope tests where
the expected isDefaultValue is incorrectly specified in the test file (the actual parsing
logic in ApplicationManifestParser for isDefaultValue is already correct).

Canonical link: <a href="https://commits.webkit.org/269021@main">https://commits.webkit.org/269021@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/c32a2323acb3d1698bae38257683cc81f0b0f53e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21317 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21644 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22370 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23179 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19772 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21556 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24937 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21880 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20983 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21540 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21212 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18476 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24032 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18380 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19324 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25665 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19455 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19529 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23506 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20021 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17045 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19330 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5109 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23584 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19916 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp [hunk 1]
```diff
@@ -298,6 +325,53 @@ Vector<ApplicationManifest::Icon> ApplicationManifestParser::parseIcons(const JS
...
+        }
+
+        URL shortcutURL(m_manifestURL, urlStringValue);
+        if (shortcutURL.isEmpty())
+            continue;
+        if (!shortcutURL.isValid()) {
+            logManifestPropertyInvalidURL("url"_s);
+            continue;
+        }
+        currentShortcut.url = shortcutURL;
```
Comment (cdumez) [inline Source/WebCore/Modules/applicationmanifest/ApplicationManifestParser.cpp]: `WTFMove(shortcutURL)`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 2]
```diff
@@ -110,6 +111,30 @@ class ApplicationManifestParserTest : public testing::Test {
         return parseString(manifestContent);
     }
 
+    ApplicationManifest parseShortcutFirstTopLevelProperty(const String& key, const String& value)
+    {
+        String manifestContent = "{ \"shortcuts\": [{\"" + key + "\": " + value + ", \"url\": \"example\" }]}";
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Please use makeString() and ASCIILiteral, like so:
`auto manifestContent = makeString("{ \"shortcuts\": [{\""_s, key, "\": "_s, value, ", \"url\": \"example\" }]}"_s);`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 3]
```diff
@@ -110,6 +111,30 @@ class ApplicationManifestParserTest : public testing::Test {
...
+        return parseString(manifestContent);
+    }
+
+    ApplicationManifest parseShortcutFirstTopLevelPropertyForURL(const String& key, const String& value)
+    {
+        String manifestContent = "{ \"shortcuts\": [{\"" + key + "\": " + value + " }]}";
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: ditto

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 4]
```diff
@@ -110,6 +111,30 @@ class ApplicationManifestParserTest : public testing::Test {
...
+        return parseString(manifestContent);
+    }
+
+    ApplicationManifest parseShortcutIconFirstTopLevelProperty(const String& key, const String& value)
+    {
+        String manifestContent = "{ \"shortcuts\": [{\"url\": \"example\", \"icons\": [{\"" + key + "\": " + value + ", \"src\": \"icon/example.png\" }]}]}";
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: ditto

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 5]
```diff
@@ -110,6 +111,30 @@ class ApplicationManifestParserTest : public testing::Test {
...
+        return parseString(manifestContent);
+    }
+
+    ApplicationManifest parseShortcutIconFirstTopLevelPropertyForSrc(const String& key, const String& value)
+    {
+        String manifestContent = "{ \"shortcuts\": [{\"url\": \"example\", \"icons\": [{\"" + key + "\": " + value + " }]}]}";
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: ditto

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 6]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
         EXPECT_EQ(expectedValue, value);
     }
 
+    void testCategories(const Vector<String>& expectedValue)
+    {
+        StringBuilder builder;
+        builder.append("[");
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: `'['`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 7]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
...
+        for (size_t i = 0; i < expectedValue.size(); i++) {
+            builder.append("\"");
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: `'\|'`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 8]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
...
+            builder.append(expectedValue[i]);
+            builder.append("\"");
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: `'\"'`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 9]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
         EXPECT_EQ(expectedValue, value);
     }
 
+    void testCategories(const Vector<String>& expectedValue)
+    {
+        StringBuilder builder;
+        builder.append("[");
+        for (size_t i = 0; i < expectedValue.size(); i++) {
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: I would use a modern for loop:
```
for (auto& value : expectedValue) {
    if (builder.length() > 1)
        builder.append(", "_s);
```

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 10]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
...
+            builder.append("\"");
+            builder.append(expectedValue[i]);
+            builder.append("\"");
+            if (i < expectedValue.size() - 1)
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Not needed with my suggestion above.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 11]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
         EXPECT_EQ(expectedValue, value);
     }
 
+    void testCategories(const Vector<String>& expectedValue)
+    {
+        StringBuilder builder;
+        builder.append("[");
+        for (size_t i = 0; i < expectedValue.size(); i++) {
+            builder.append("\"");
+            builder.append(expectedValue[i]);
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: `builder.append(value);` with my suggestion above.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 12]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
...
+            builder.append("\"");
+            if (i < expectedValue.size() - 1)
+                builder.append(", ");
+        }
+        builder.append("]");
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]:
```
builder.append(']');
```

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 13]
```diff
@@ -207,18 +232,45 @@ class ApplicationManifestParserTest : public testing::Test {
...
+        auto categoriesValue = builder.toString();
+        String manifestContent = "{ \"categories\" : " + categoriesValue + " }";
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Please use makeString() and ASCIILiteral (`_s`)

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm [hunk 14]
```diff
@@ -77,6 +77,22 @@
     };
 }
 
+static std::optional<WebCore::ApplicationManifest::Shortcut> makeVectorElement(const WebCore::ApplicationManifest::Shortcut*, id arrayElement)
```
Comment (cdumez) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:78]: Where is this function used?

**Review by cdumez — APPROVED** [review]
LGTM with nits.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h [hunk 15]
```diff
@@ -72,7 +73,9 @@ WK_CLASS_AVAILABLE(macos(10.13.4), ios(11.3))
 @property (nonatomic, readonly, copy) NSURL *startURL;
 @property (nonatomic, readonly, copy) NSURL *manifestId WK_API_AVAILABLE(macos(13.3), ios(16.4));
 @property (nonatomic, readonly) _WKApplicationManifestDisplayMode displayMode;
+@property (nonatomic, readonly, copy) NSArray<NSString *> *categories WK_API_AVAILABLE(macos(14.2), ios(17.2));
```
Comment (szewai) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h]: Please use WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h [hunk 16]
```diff
@@ -72,7 +73,9 @@ WK_CLASS_AVAILABLE(macos(10.13.4), ios(11.3))
...
 @property (nonatomic, readonly, copy) NSArray<_WKApplicationManifestIcon *> *icons WK_API_AVAILABLE(macos(13.0), ios(16.0));
+@property (nonatomic, readonly, copy) NSArray<_WKApplicationManifestShortcut *> *shortcuts WK_API_AVAILABLE(macos(14.2), ios(17.2));
```
Comment (szewai) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h]: Ditto

**Review by szewai — CHANGES_REQUESTED** [review]

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm (hunk 14 above)
Comment (LumingYin) [reply Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm:78]: This is used later in line 259 in -initWithCoder::
`makeVector<WebCore::ApplicationManifest::Shortcut>(shortcuts),`

Implemented in VectorCocoa.h, `makeVector` is a template that calls into `makeVectorElement`.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp (hunk 2 above)
Comment (LumingYin) [reply Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Switched to using `makeString()` and ASCIILiteral for all newly added methods.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp (hunk 7 above)
Comment (LumingYin) [reply Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Switched to `'\"'`.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp [hunk 17]
```diff
@@ -207,18 +232,47 @@ class ApplicationManifestParserTest : public testing::Test {
...
+    {
+        StringBuilder builder;
+        builder.append('[');
+        auto expectedValueSize = expectedValue.size();
+        for (auto& value : expectedValue) {
+            builder.append('\"');
+            builder.append(value);
+            builder.append('\"');
+            auto builderLength = builder.length();
+            if (builderLength > 1 && builderLength < expectedValueSize)
```
Comment (cdumez) [inline Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: This isn't what I had suggested. Why the change?

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp (hunk 9 above)
Comment (LumingYin) [reply Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Switched to a modern for loop.

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp (hunk 17 above)
Comment (cdumez) [reply Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: My proposal was to move it to *before* we append the quoted value and with only one `builderLength > 1` being checked. What you ended up with doesn't make sense. `builderLength > 1` is always true in your case.

**Review by cdumez — CHANGES_REQUESTED** [review]
builder code needs fixing.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm [hunk 18]
```diff
@@ -77,6 +77,22 @@
...
+{
+    if (![arrayElement isKindOfClass:_WKApplicationManifestShortcut.class])
+        return std::nullopt;
+
```
Comment (whsieh) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm]: Nit - you can omit this early return, since `dynamic_objc_cast` will return `nil` below if the class check fails.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h (hunk 15 above)
Comment (LumingYin) [reply Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h]: Switched to `WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA))` for both.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm [hunk 19]
```diff
@@ -137,6 +153,62 @@ - (void)dealloc
...
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+    if (!(self = [self init]))
+        return nil;
+
+    _name = [[coder decodeObjectOfClass:[NSString class] forKey:@"name"] copy];
+    _url = [[coder decodeObjectOfClass:[NSURL class] forKey:@"url"] copy];
+    _icons = [[coder decodeObjectOfClasses:[NSSet setWithArray:@[[NSArray class], [_WKApplicationManifestIcon class], [NSString class]]] forKey:@"icons"] copy];
+
```
Comment (whsieh) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm]: I would recommend using `RetainPtr` instead of manual `-retain` / `-release`

File: Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp (hunk 17 above)
Comment (LumingYin) [reply Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp]: Apologies! I misunderstood your suggestion and didn't realize it's meant to be moved before we append the quote. Just updated the PR to move it before we append the quote and drop the unnecessary `builderLength < expectedValueSize` check.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm (hunk 18 above)
Comment (LumingYin) [reply Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm]: Removed this early return.

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm (hunk 19 above)
Comment (LumingYin) [reply Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.mm]: Switched to using `RetainPtr`, thank you for the patch!

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h [hunk 20]
```diff
@@ -100,4 +103,13 @@ WK_CLASS_AVAILABLE(macos(13.0), ios(16.0))
 
 @end
 
+WK_CLASS_AVAILABLE(macos(14.2), ios(17.2))
```
Comment (cdumez) [inline Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h]: Isn't this supposed to be TBA?

**Review by cdumez — APPROVED** [review]
LGTM with question

File: Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h (hunk 20 above)
Comment (LumingYin) [reply Source/WebKit/UIProcess/API/Cocoa/_WKApplicationManifest.h]: Yes! Replaced this with TBA.

**Review by szewai — APPROVED** [review]

**Review by whsieh — APPROVED** [review]
